odoo/addons/test_orm/tests/test_schema.py

```diff
@@ -226,519 +88,443 @@ def test_models_fields(self):
-        """ check the database representation of a float field """
-        model = self.env['test_orm.mixed']
-        columns_data = self.get_columns_data(model._table)
-        self.assertEqual(columns_data['number'], {
-            'character_maximum_length': None,
-            'column_default': None,
-            'column_name': 'number',
-            'data_type': 'numeric',
-            'datetime_precision': None,
-            'is_nullable': 'YES',
-            'is_updatable': 'YES',
-            'numeric_precision': None,
-            'numeric_precision_radix': 10,
-            'numeric_scale': None,
-            'table_catalog': self.cr.dbname,
-            'table_name': 'test_orm_mixed',
-            'table_schema': 'public',
-            'udt_catalog': self.cr.dbname,
-            'udt_name': 'numeric',
-            'udt_schema': 'pg_catalog',
-        })
+        self.assertEqual(
+            self._get_table_data(table),
+            self._expected_table_data(),
+        )
 
-    def test_10_monetary(self):
-        """ check the database representation of a monetary field """
-        model = self.env['test_orm.mixed']
-        columns_data = self.get_columns_data(model._table)
-        self.assertEqual(columns_data['amount'], {
-            'character_maximum_length': None,
-            'column_default': None,
-            'column_name': 'amount',
-            'data_type': 'numeric',
-            'datetime_precision': None,
-            'is_nullable': 'YES',
-            'is_updatable': 'YES',
-            'numeric_precision': None,
-            'numeric_precision_radix': 10,
-            'numeric_scale': None,
-            'table_catalog': self.cr.dbname,
-            'table_name': 'test_orm_mixed',
-            'table_schema': 'public',
-            'udt_catalog': self.cr.dbname,
-            'udt_name': 'numeric',
-            'udt_schema': 'pg_catalog',
-        })
-
-    def test_10_char(self):
-        """ check the database representation of a char field """
-        model = self.env['res.country']
-        self.assertTrue(type(model).code.required)
-        self.assertEqual(type(model).code.size, 2)
-        columns_data = self.get_columns_data(model._table)
-        self.assertEqual(columns_data['code'], {
-            'character_maximum_length': 2,
-            'column_default': None,
-            'column_name': 'code',
-            'data_type': 'character varying',
-            'datetime_precision': None,
-            'is_nullable': 'NO',
-            'is_updatable': 'YES',
-            'numeric_precision': None,
-            'numeric_precision_radix': None,
-            'numeric_scale': None,
-            'table_catalog': self.cr.dbname,
-            'table_name': 'res_country',
-            'table_schema': 'public',
-            'udt_catalog': self.cr.dbname,
-            'udt_name': 'varchar',
-            'udt_schema': 'pg_catalog',
-        })
+        self.assertEqual(
+            set(self.columns_data.keys()), {
```

Comment:
Use `sorted()` on the keys, and compare it to a (sorted) list.  This is going to be easier to debug if that one is broken.